src/components/pixi/viewport/plugins/markingModePlugin.ts

```diff
@@ -0,0 +1,203 @@
+            new Event(CUSTOM_GLOBAL_EVENTS.INTERRUPT_MARKING)
+        );
+        this.removeEventListeners();
+        this.currentHandler = null;
+    }
+
+    private isMarkingModeActive(): boolean {
+        return (
+            DashboardToolbarStore.state.settings.cursor.mode ===
+            CURSOR_MODES.MARKING
+        );
+    }
+
+    private shouldHandleMarking(): boolean {
+        return this.isMarkingModeActive() && !this.spacePressed;
+    }
+
+    private handleMouseDown = (e: FederatedPointerEvent): void => {
+        if (this.shouldHandleMarking() && e.button === 0) {
+            this.startMarking(e);
+        }
+    };
+
+    private handleRightDown = (e: FederatedPointerEvent): void => {
+        if (this.shouldHandleMarking() && IS_DEV_ENVIRONMENT) {
+            console.log(e);
+        }
+    };
+
+    private handleKeyDown = (e: KeyboardEvent): void => {
+        if (e.code === "Space") this.spacePressed = true;
+    };
+
+    private handleKeyUp = (e: KeyboardEvent): void => {
+        if (e.code === "Space") this.spacePressed = false;
+    };
+
+    private startMarking(e: FederatedPointerEvent): void {
+        const { viewport, markingsStore } = this.handlerParams;
+        if (
+            viewport.children.length < 1 ||
+            markingsStore.state.temporaryMarking !== null
+        )
+            return;
+
+        document.addEventListener(
+            CUSTOM_GLOBAL_EVENTS.INTERRUPT_MARKING,
+            this.handleInterrupt
+        );
+
+        const { markingClass } = DashboardToolbarStore.state.settings.marking;
+        const workingMode = WorkingModeStore.state.workingMode!;
+        const { id: characteristicId } =
+            MarkingCharacteristicsStore.actions.activeCharacteristics.getActiveCharacteristicByMarkingClass(
+                markingClass,
+                workingMode
+            );
+
+        switch (markingClass) {
+            case MARKING_CLASS.POINT:
+                this.currentHandler = new PointMarkingHandler(
+                    this,
+                    characteristicId,
+                    e
+                );
+                break;
+            case MARKING_CLASS.RAY:
+                this.currentHandler = new RayMarkingHandler(
+                    this,
+                    characteristicId,
+                    e
+                );
+                break;
+            case MARKING_CLASS.LINE_SEGMENT:
+                this.currentHandler = new LineSegmentMarkingHandler(
+                    this,
+                    characteristicId,
+                    e
+                );
+                break;
+            case MARKING_CLASS.BOUNDING_BOX:
+                this.currentHandler = new BoundingBoxMarkingHandler(
+                    this,
+                    characteristicId,
+                    e
+                );
+                break;
+            default:
+                throw new Error(`Unsupported marking class: ${markingClass}`);
+        }
```

Comment:
```suggestion
        const MARKING_HANDLERS = {
            [MARKING_CLASS.POINT]: PointMarkingHandler,
            [MARKING_CLASS.RAY]: RayMarkingHandler,
            [MARKING_CLASS.LINE_SEGMENT]: LineSegmentMarkingHandler,
            [MARKING_CLASS.BOUNDING_BOX]: BoundingBoxMarkingHandler
        };

        const HandlerClass = MARKING_HANDLERS[markingClass];

        if (!HandlerClass) {
            throw new Error(`Unsupported marking class: ${markingClass}`);
        }

        this.currentHandler = new HandlerClass(this, characteristicId, e);
```

Will this code do the job? it's much simpler as the arguments are the same between them.

Reply:
I have to agree with Marcel here; the suggested code is much more readable and compact.

Reply:
If it works, I didn't test it. 🫠